api/messages.go

```diff
@@ -28,14 +29,43 @@ func (br *decodingError) Error() string {
 }
 
 type CreateCredentialRequest struct {
-	Address string `json:"address"`
-	Msg     string `json:"msg"`
-	Sig     string `json:"sig"`
-	Version string `json:"version"`
+	Address common.Address `json:"address"`
+	Msg     []byte         `json:"msg"`
+	Sig     []byte         `json:"sig"`
+	Version string         `json:"version"`
 
 	operatorType credentials.OperatorType `json:"-"`
 }
 
+func (c *CreateCredentialRequest) UnmarshalJSON(data []byte) error {
+	type Alias CreateCredentialRequest
+	aux := &struct {
+		Address string `json:"address"`
+		Msg     string `json:"msg"`
+		Sig     string `json:"sig"`
+
+		// Populates the `Version` field
+		*Alias
+	}{
+		Alias: (*Alias)(c),
+	}
+
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	// Convert Address
+	c.Address = common.HexToAddress(aux.Address)
+
+	// Convert Msg
+	c.Msg = []byte(aux.Msg)
+
+	// Convert Sig
+	c.Sig = []byte(aux.Sig)
```

Comment:
need to parse hex here- you can use https://pkg.go.dev/github.com/ethereum/go-ethereum@v1.14.11/common/hexutil#Decode

Reply:
updated and pushed